lib/foreman_maintain/package_manager/dnf.rb

```diff
@@ -1,10 +1,44 @@
 module ForemanMaintain::PackageManager
   class Dnf < Yum
+    PROTECTOR_CONFIG_FILE = '/etc/yum/pluginconf.d/foreman-protector.conf'.freeze
+    PROTECTOR_WHITELIST_FILE = '/etc/yum/pluginconf.d/foreman-protector.whitelist'.freeze
+
     def clean_cache(assumeyes: false)
       dnf_action('clean', 'all', :assumeyes => assumeyes)
       super
     end
 
+    def dnf_plugin_dirs
+      dnf_plugin_parent_dirs = Dir.glob('/usr/lib/python3*').map { |d| d + '/site-packages' }
+      dnf_plugin_parent_dirs.map do |d|
+        if Dir.new(d).children.include?('dnf-plugins')
+          d + '/dnf-plugins'
+        end
+      end.compact
+    end
+
+    def protector_plugin_files
+      @protector_plugin_files ||= dnf_plugin_dirs.map do |d|
+        d + '/' + 'foreman-protector.py'
+      end
+    end
+
+    def install_version_locking(*)
+      protector_plugin_files.each do |f|
+        install_extras('foreman_protector/dnf/foreman-protector.py', f)
+      end
+      super
+    end
```

Comment:
I'm still struggling to understand, why these steps are needed at all.

The files are installed there by packaging correctly already (excerpt from EL7, but otherwise correct):
```
# rpm -ql rubygem-foreman_maintain |grep prote
/etc/yum/pluginconf.d/foreman-protector.conf
/etc/yum/pluginconf.d/foreman-protector.whitelist
/usr/lib/yum-plugins/foreman-protector.py
/usr/lib/yum-plugins/foreman-protector.pyc
/usr/lib/yum-plugins/foreman-protector.pyo
```

Moreso, I don't think tools should poke around in `/usr/lib` if they are not `rpm` themself.

I this solely for the case where f-m is installed from source/gem, vs packaging?

Reply:
That is one reason and another could be if these files are missing or somehow deleted then there should be way to restore it without reinstalling rpm? I do think reinstalling rpm should be ideal way

Reply:
With the same logic, they could be missing from `/usr/share/gems/gems/foreman_maintain-1.0.3/extras/foreman_protector` where they are copied from too ;)

And yes, reinstalling the package should be the way to get the files back.

Reply:
I have removed the old way of installing locking tooling and now implemented reinstall of the package if any of the files are missing. Putting dnf-plugin specific python file at correct location will be handled by packaging pr.

Let me know what you think